pkg/controller/instanceset/utils.go

```diff
@@ -249,3 +253,62 @@ func GetPodNameSetFromInstanceSetCondition(its *workloads.InstanceSet, condition
 	}
 	return podSet
 }
+
+// CalculateConcurrencyReplicas returns absolute value of concurrency for workload. This func can solve some
+// corner cases about percentage-type concurrency, such as:
+// - if concurrency > "0%" and replicas > 0, it will ensure at least 1 pod is reserved.
+// - if concurrency < "100%" and replicas > 1, it will ensure at least 1 pod is reserved.
+//
+// if concurrency is nil, concurrency will be treated as 100%.
+func CalculateConcurrencyReplicas(concurrency *intstr.IntOrString, replicas int) (int, error) {
```

Comment:
Any difference with `intstr.GetScaledValueFromIntOrPercent`?

Reply:
Oh Sorry, This function doesn't exist in the old version of the dependency, but I forgot that KB's dependency was a higher version.